File: codinGame/Python3/power_of_thor.py
lx, ly, x, y = [int(i) for i in input().split()]

while True:
    move  = 'N' if y>ly else 'S' if y<ly else ''
    move += 'W' if x>lx else 'E' if x<lx else ''
    
    if 'N' in move: y-=1
    if 'S' in move: y+=1
    if 'W' in move: x-=1
    if 'E' in move: x+=1

    print(move)

File: codinGame/Perl/power_of_thor.pl
use strict;
use warnings;
use 5.32.1;

select(STDOUT); $| = 1;

chomp(my $input = <STDIN>);
my ($lx, $ly, $tx, $ty) = split / /, $input;

while (<STDIN>) {
    print(($ty>$ly?'N':$ty<$ly?'S':'').($tx>$lx?'W':$tx<$lx?'E':'')."\n");
    $tx += ($lx>$tx) - ($lx<$tx);
    $ty += ($ly>$ty) - ($ly<$ty);
}
